packages/google_maps_flutter/google_maps_flutter/ios/Classes/GoogleMapHeatmapController.m

```diff
@@ -0,0 +1,159 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "GoogleMapHeatmapController.h"
+#import "JsonConversions.h"
+@import GoogleMapsUtils;
+
+@implementation FLTGoogleMapHeatmapController {
+  GMUHeatmapTileLayer *_heatmap;
+  GMSMapView *_mapView;
+}
+- (instancetype)initWithHeatmap:(GMUHeatmapTileLayer *)heatmap mapView:(GMSMapView *)mapView {
+  self = [super init];
+  if (self) {
+    _heatmap = heatmap;
+    _mapView = mapView;
+
+    // Workaround for weird heatmap rendering compared to other platforms
+    // Context:
+    // https://github.com/googlemaps/google-maps-ios-utils/issues/143#issuecomment-343850029
+    _heatmap.minimumZoomIntensity = 0;
+    _heatmap.maximumZoomIntensity = 21;
+  }
+  return self;
+}
+
+- (void)removeHeatmap {
+  _heatmap.map = nil;
+}
+
+- (void)clearTileCache {
+  [_heatmap clearTileCache];
+}
+
+#pragma mark - FLTGoogleMapHeatmapOptionsSink methods
+
+- (void)setWeightedData:(NSArray<GMUWeightedLatLng *> *)weightedData {
+  _heatmap.weightedData = weightedData;
+}
+
+- (void)setGradient:(GMUGradient *)gradient {
+  _heatmap.gradient = gradient;
+}
+
+- (void)setOpacity:(double)opacity {
+  _heatmap.opacity = opacity;
+}
+
+- (void)setRadius:(int)radius {
+  _heatmap.radius = radius;
+}
+
+- (void)setMap {
+  _heatmap.map = _mapView;
+}
+@end
+
+static int ToInt(NSNumber *data) { return [FLTGoogleMapJsonConversions toInt:data]; }
```

Comment:
These "ToX" static functions are essentially calling other OBJC class methods directly. It seems to be the pattern that we have been using in many places in this plugin. I'd suggest not doing this anymore as it is not necessary and creates confusions..

Reply:
Should we change all of them?

Reply:
Yes please :)

Reply:
I meant in all of the other controllers, but based on your other comments I'll just change them in this one.

Reply:
Done